stub.c

```diff
@@ -106,27 +106,34 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table)
 		bt_cmdline = (CHAR8 *)(loaded_image->ImageBase + addrs[0]);
 
 	bt_cmdline_len = szs[0];
+	if (bt_cmdline[bt_cmdline_len] != '\0') {
```

Comment:
Sorry I suspect the answer is in pe_memory_locate_sections(), but - is bt_cmdline bt_cmdline_len+1 chars long?

Reply:
hm.. that is a good question. this is the builtin command line. so its what was `objcopy`ed in.

Reply:
I like knowing that it is a null terminated string.

You're suggesting that I could potentially be reading outside of the memory and cause segfault or something, right?

Reply:
Yes, I'm suggesting bt_cmdline may have been only allocated to bt_cmdline_len chars.  I'll dig deeper in a few mins to see if I can find that out

Reply:
Wait - note also that bt_cmdline is only assigned if szs[0] > 0, but you in any case chekc bt_cmdline[bt_cmdline_len].  So if szs[0] <= 0 then this is a guaranteed NULL deref.